lib/spack/spack/fetch_strategy.py

```diff
@@ -392,13 +392,31 @@ def __init__(self, *args, **kwargs):
 
     @_needs_stage
     def fetch(self):
-        super(CacheURLFetchStrategy, self).fetch()
+        path = re.sub('^file://', '', self.url)
+
+        # check whether the cache file exists.
+        if not os.path.isfile(path):
+            raise NoCacheError('No cache of %s' % path)
+
+        self.stage.chdir()
+
+        # remove old symlink if one is there.
+        filename = self.stage.save_filename
+        if os.path.exists(filename):
+            os.remove(filename)
+
+        # Symlink to local cached archive.
+        os.symlink(path, filename)
+
+        # Notify the user how we fetched.
+        tty.msg('Using cached archive: %s' % path)
+
+        # Remove link if checksum fails, or subsequent fetchers
+        # will assume they don't need to download.
         if self.digest:
             try:
                 self.check()
             except ChecksumError:
-                # Future fetchers will assume they don't need to
-                # download if the file remains
                 os.remove(self.archive_file)
```

Comment:
In short: it doesn't need to. Both save_filename and archive_file pull from "expected_archive_files". save_filename picks one as preferred. archive_file looks for whatever actually exists